pkg/scheduler/framework/session.go

```diff
@@ -279,6 +297,46 @@ func (ssn *Session) Allocate(task *api.TaskInfo, hostname string) error {
 	}
 
 	if ssn.JobReady(job) {
+
+		// patch all the pod with backfill annotation. if any fails, abort the backfill
+		var wg sync.WaitGroup
+		errChannel := make(chan error, 1)
+		finished := make(chan bool, 1)
+
+		for _, pendingTask := range job.TaskStatusIndex[api.Allocated] {
+			if pendingTask.Condition.IsBackfill {
+
+				wg.Add(1)
+				go func(pendingTask *api.TaskInfo) {
+					defer wg.Done()
+
+					annotation := map[string]string{v1alpha1.BackfillAnnotationKey: "true"}
+					err := ssn.PatchAnnotation(pendingTask, annotation)
+					if err != nil {
+						glog.Errorf("Failed to patch backfill=true to task/pod <%s/%s>: %v",
+							pendingTask.Name, pendingTask.Pod.Name, err)
+						errChannel <- err
+						return
+					}
+				}(pendingTask)
+			}
+		}
+
+		go func() {
+			wg.Wait()
+			close(finished)
+		}()
+
+		select {
+		case <-finished:
+		case err := <-errChannel:
+			if err != nil {
+				fmt.Println("error ", err)
+				return err
```

Comment:
What happen if Pod is patched but did not dispatch because of other error?

Reply:
If dispatch fails, the pod remains unbound in the current scheduling round, and therefore will not be started. In the next scheduling round, when we create TaskInfo for a Pod in `NewTaskInfo`, we will noticed that the Pod is unbound but has backfill annotation. In this case, we will remove backfill annotation from the Pod. 

@pdgetrf It looks to me that we do not yet have the logic to remove backfill annotation in `NewTaskInfo`.

Reply:
>  In this case, we will remove backfill annotation from the Pod.

It seems not implemented in this PR, right?

Reply:
this can be fixed in a following up PR which, as you already suggested, is smaller and easier to review. And I do not see how this should block the current PR. It will only make this one even bigger.

Reply:
> What happen if Pod is patched but did not dispatch because of other error?

code has the answer.